chia/full_node/full_node_api.py

```diff
@@ -1273,13 +1273,32 @@ async def respond_compact_proof_of_time(self, request: timelord_protocol.Respond
     @execute_task
     @peer_required
     @api_request
-    async def new_compact_vdf(self, request: full_node_protocol.NewCompactVDF, peer: ws.WSChiaConnection):
+    @bytes_required
+    async def new_compact_vdf(
+        self, request: full_node_protocol.NewCompactVDF, peer: ws.WSChiaConnection, request_bytes: bytes = b""
+    ):
         if self.full_node.sync_store.get_sync_mode():
             return None
+
+        if len(self.full_node.compact_vdf_sem._waiters) > 20:
+            self.log.debug(f"Ignoring NewCompactVDF: {request}, _waiters")
+            return
+
+        name = std_hash(request_bytes)
+        if name in self.full_node.compact_vdf_requests:
+            self.log.debug(f"Ignoring NewCompactVDF: {request}, already requested")
+            return
+        self.full_node.compact_vdf_requests.add(name)
+
         # this semaphore will only allow a limited number of tasks call
         # new_compact_vdf() at a time, since it can be expensive
         async with self.full_node.compact_vdf_sem:
-            await self.full_node.new_compact_vdf(request, peer)
+            try:
+                await self.full_node.new_compact_vdf(request, peer)
+            except BaseException as e:
+                raise e
```

Comment:
what's the behaviour if we raise the exception, are we going to disconnect the peer? how about a log.error? this looks good to me otherwise.

Reply:
it's just re-raising, so default behavior is not changed. Peer gets disconnected, all task from that peer stopped.